### ISSUE TITLE
RPA and NORPA create exception even though not called together

### ISSUE THREAD

**Issue by Noordsestern** [robotframework/robotframework#4360]
# Challenge
I have robot tasks with keyword resource that I would like to test with robot test cases. Thus, I have task suites and test suites in the same project. Imagine this simple project:

```
/projectfolder
--> tasks.robot
--> tests.robot
```

With `tasks.robot`
```
*** Tasks ***
My task
    Log To Console     I am task
```
and `tests.robot`
```
*** Test Cases ****
My test
    Log To Console    I am test
```

Now when I call either the test or the task like this, I get an exception:
`robot -t "My test" .`

The exception is that I would be mixing tasks and test cases.

I am grateful that I cannot mix tasks and test cases in one execution. When collecting all test cases to be executed, there should not be any rpa-task be available that could be mixed here.

# Proposal
Do rpa/norpa-mix evaluation later. If I for some reason made a mistake by calling a task "My test" I would still be happy to get the exception. But if I already make specific restrictions (suite name, test name, included tags...) that only point to either rpa or norpa executions, the exception should not come.

# Use Case
- Testing Robot Tasks with Robot Tests in pipeline
- Plugins for Robot Framework that call complete folders and break, when pre-parsing finds Tasks and Test Cases in same project

# Workaround
I can of course call the robot file directly instead of a folder. But that is only a workaround and is prone to file-movement and renaming of files. And it does not help with plugins relying on the parser, that throughs exception even though no rpa-nopa-mix occured.

**Comment by d-biehl** [comment]
RobotCode uses RF's TestSuiteBuilder (with a few small extensions) to find the available test cases/tasks, so I have the issue in RobotCode as well. 

And I regularly get issues on this topic, which I then answer by pointing to RobotFramework documentation and saying turn on the right mode (RPA/NOROA). 

But the question I then ask myself is and I get asked: Why is it not possible to mix tests and tasks? 
There is no difference at runtime, only when the developer/tester creates tests/tasks and when writing the log/reports. And if you set the mode to RPA or NORPA, tests become tasks and the other way around.

Wouldn't it make more sense to have a mode where mixing tests and tasks would be possible and this would also be taken into account in reporting?

Alternatively or additionally we could introduce a nostrict-rpa/nostrict-norpa mode in which the error "Conflicting execution modes" is not thrown but the conflicting test/task is simply filtered.

**Comment by Noordsestern** [comment]
Might this be fixed with #4807 ?

**Comment by pekkaklarck** [comment]
Yes, this is effectively a duplicate of #4807.
